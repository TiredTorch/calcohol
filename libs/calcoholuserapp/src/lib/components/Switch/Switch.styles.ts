export const switchStyles = {
	root: {
		"& .MuiSwitch-thumb": {
			background: "#8449D0"
		}
	}
};