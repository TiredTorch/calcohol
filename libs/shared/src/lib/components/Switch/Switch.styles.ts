export const switchStyles = {
	root: {
		width: "100%",
		display: "flex",
		alignItems: "center",
		justifyContent: "space-around"
	},
	text: {
		width: "35%",
		textAlign: "center"
	}
};