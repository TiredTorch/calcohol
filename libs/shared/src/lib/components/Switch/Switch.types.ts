import { SwitchProps as MUISwitchProps } from "@mui/material";

export type SwitchProps = MUISwitchProps & {
	leftLable: string;
	rightLable: string;
};