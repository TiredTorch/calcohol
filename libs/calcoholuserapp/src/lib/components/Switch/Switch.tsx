import { FC } from "react";
import { Switch as MUISwitch, SwitchProps } from "@mui/material";
import { switchStyles } from "./Switch.styles";

export const Switch: FC<SwitchProps> = ({
	...rest
}) => {
	return (
		<MUISwitch
			sx={switchStyles.root}
			{...rest}
		/>
	);
};

export default Switch;