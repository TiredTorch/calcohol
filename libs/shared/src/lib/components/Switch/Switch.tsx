import { FC } from "react";
import { Box, Switch as MUISwitch, Typography } from "@mui/material";
import { SwitchProps } from "./Switch.types";
import { switchStyles } from "./Switch.styles";

export const Switch: FC<SwitchProps> = ({
	leftLable,
	rightLable,
	...rest
}) => {
	return (
		<Box
			component="div"
			sx={switchStyles.root}
		>
			<Typography
				variant="h2"
				sx={switchStyles.text}
			>
				{leftLable}

			</Typography>
			<MUISwitch
				{...rest}
			/>
			<Typography
				variant="h2"
				sx={switchStyles.text}
			>
				{rightLable}

			</Typography>
		</Box>
	);
};

export default Switch;