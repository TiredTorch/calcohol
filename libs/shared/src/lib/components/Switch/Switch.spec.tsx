import "@testing-library/jest-dom";
import { render, queryByText } from "@testing-library/react";
import { faker } from "@faker-js/faker";
import { Switch } from "./Switch";

afterEach(() => {
	jest.resetAllMocks();
});

const mockedLeftLable = faker.datatype.string(10);
const mockedRightLable = faker.datatype.string(10);

it("should render switch", () => {
	const { baseElement } = render(
		<Switch
			leftLable={mockedLeftLable}
			rightLable={mockedRightLable}
		/>
	);

	expect(baseElement).toBeInTheDocument();
	expect(queryByText(baseElement, mockedLeftLable)).toBeInTheDocument();
	expect(queryByText(baseElement, mockedRightLable)).toBeInTheDocument();
	expect(baseElement).toMatchSnapshot();
});