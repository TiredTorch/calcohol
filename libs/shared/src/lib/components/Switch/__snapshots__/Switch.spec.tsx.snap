// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render switch 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-18m6hx6"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-uw06uf-MuiTypography-root"
      >
        j%l.ISxo!+
      </h2>
      <span
        class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
        >
          <input
            class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            type="checkbox"
          />
          <span
            class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <h2
        class="MuiTypography-root MuiTypography-h2 css-uw06uf-MuiTypography-root"
      >
        _iMl\\"i.l%
      </h2>
    </div>
  </div>
</body>
`;
